### ISSUE TITLE
`shortpath` differences with default project

### ISSUE THREAD

**Issue by jo-basevi** [payu-org/payu#502]
There's a difference between `shortpath` default on the login node and pbs payu run submission when `project` is defined in `config.yaml`.  With the logic with creating new `UUID`s when there isn't a pre-existing archive, this can result in restarts not being picked up in payu as brought up in this forum post: https://forum.access-hive.org.au/t/payu-archives-and-restarts/3626

To replicate the error on my NCI account, I had a `config.yaml` with the following settings:

```yaml
project: nf33
# shortpath: /scratch/nf33
```

Then running payu commands on the login node (e.g. `payu setup`/`payu checkout -b`/`payu clone`), the `shortpath` defaults to `/scratch/tm70` because `tm70` is my default project. This creates an archive at `/scratch/nf33/jb4202/mom6/archive/mom6-double-grye-base-c248b389`. Then on `payu run`, the`shortpath` defaults to `/scratch/nf33` and there isn't an archive under `/scratch/nf33/jb4202/mom6/archive/mom6-double-grye-base-c248b389` so it creates a new archive with a new UUID, e.g. `/scratch/nf33/jb4202/mom6/work/mom6-double-grye-base-42bdfc18`. Then once the run has finished, running `payu setup`, it can't find `/scratch/tm70/jb4202/mom6/work/mom6-double-grye-base-42bdfc18` so it creates a new UUID and archive, and so on..

So basically payu needs to use the `project` in `config.yaml` in it's default project for `shortpath`. `shortpath` is currently being set here: 
https://github.com/payu-org/payu/blob/2826621a83bcb13315d3a003a946c5ca8069b1cd/payu/laboratory.py#L69-L74

I think sometime like the following might work:

```
default_project = config.get('project', os.environ.get('PROJECT', ''))
```

In the meantime to avoid this bug, setting `shortpath` to the `config.yaml` to match the project would work. For example:

```yaml
project: nf33
shortpath: /scratch/nf33
```

**Comment by aidanheerdegen** [comment]
Nice detective work!
